### PR TITLE
Updates eslint and centralizes some config options.

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @philipwalton @addyosmani @gauntface

This is part of #478, but it only attempts to normalize the usage of per-directory `.eslintrc*` files. It doesn't attempt to do anything about the use of inline `/* eslint-disable */` and similar rules. That will be done in a follow-up PR.

### CODE REVIEW

File: packages/workbox-runtime-caching/test/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  globals: {
```
Comment (philipwalton) [inline packages/workbox-runtime-caching/test/.eslintrc.js]: Did you mean to add this? I thought you wanted to remove all but the base eslintrc file.

I tend to prefer declaring globals via inline eslint comments on a per-file basis (and the only globals declared in eslintrc files are globals that apply generally to the entire project).

That way when you see a variable in a file, it's very clear that the intent is to make it global, and you don't have to look up the directory structure for eslintrc files. WDYT?

File: packages/workbox-background-sync/src/lib/request-queue.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ class RequestQueue {
         id: hash,
         url: request.url,
       });
-    } catch(e) {
+    } catch (e) {
```
Comment (philipwalton) [inline packages/workbox-background-sync/src/lib/request-queue.js]: Somewhat tangental, but in the future I'd love to see us standardize on `err` for `catch` blocks unless there's a good reason to use a more descriptive name. Maybe we could write an eslint plugin to enforce this...

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    ecmaVersion: 2017,
+    sourceType: 'module',
+  },
+  globals: {
+    BroadcastChannel: false,
+    WorkboxSW: false,
+    workbox: false,
+  },
+  rules: {
+    'no-tabs': 2,
```
Comment (philipwalton) [inline .eslintrc.js]: I think this is an oversight in `eslint-config-google`, as Google JS Style definitely requires spaces. I'll update that library (related: https://github.com/google/eslint-config-google/issues/40).

Comment (philipwalton) [inline .eslintrc.js]: If you upgrade to 0.8.1 you'll get this rule now.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    'linebreak-style': 0,
+    'space-before-function-paren': ['error', {
```
Comment (philipwalton) [inline .eslintrc.js]: I've also updated the  `eslint-config-google` rule for this, as I think it's what the Google JS style guide recommends. You can get the changes in `0.9.0`.

Also, in the future, if you're tempted to add rules/exceptions to your eslint config, feel free to ping me, as maybe it's a mistake :)

File: packages/workbox-runtime-caching/test/.eslintrc.js (hunk 1 above)
Comment (gauntface) [inline packages/workbox-runtime-caching/test/.eslintrc.js]: Personally I'm a fan of globals in eslint if all code being tested is run in that context.

File: .eslintrc.js (hunk 3 above)
Comment (jeffposnick) [inline .eslintrc.js]: Thanks!

File: packages/workbox-runtime-caching/test/.eslintrc.js (hunk 1 above)
Comment (jeffposnick) [inline packages/workbox-runtime-caching/test/.eslintrc.js]: This was a bit of a crutch as there are several files in this one directory that share a few globals, without them being used anywhere else.

There are similar issues throughout the codebase that are addressed via inline `/* eslint ... /*` config comments now, and I'm punting on addressing standardizing on that sort of thing for another PR.

**Pull request merged**